### PR TITLE
Pull out connectors

### PR DESCRIPTION
**Pull request by StarDumpling**

### CODE REVIEW

File: src/bridge/connectors/connector_0.rs [hunk 1]
```diff
@@ -18,6 +18,13 @@ pub struct Connector0 {
 
 impl Connector0 {
     pub fn new(network: Network, n_of_n_taproot_public_key: &XOnlyPublicKey) -> Self {
+        Self::new_for_validation(network, n_of_n_taproot_public_key)
+    }
+
+    pub fn new_for_validation(
```
Comment (franklynot) [inline src/bridge/connectors/connector_0.rs]: Hmm, I just did the opposite in PR #78 - I removed the last occurrences of `new_for_validation` on connectors. There is no difference in the logic between `new` and that, so I think we can just keep one of them and use it whenever we need to instantiate a connector. Whether it's for validation or when creating a new graph.

File: src/bridge/graphs/peg_in.rs [hunk 2]
```diff
@@ -115,6 +122,8 @@ pub struct PegInGraph {
     pub depositor_public_key: PublicKey,
     depositor_taproot_public_key: XOnlyPublicKey,
     depositor_evm_address: String,
+
+    connectors: PegInConnectors,
```
Comment (franklynot) [inline src/bridge/graphs/peg_in.rs]: This is adding a level of indirection whenever accessing a connector. I suggest we keep the connectors directly in the graph struct, like we do transactions.

File: src/bridge/graphs/peg_out.rs [hunk 3]
```diff
@@ -803,9 +800,16 @@ impl PegOutGraph {
         }
     }
 
-    pub fn connector_1_id(&self) -> ConnectorId { self.connector_1.id() }
-    pub fn connector_2_id(&self) -> ConnectorId { self.connector_2.id() }
-    pub fn connector_6_id(&self) -> ConnectorId { self.connector_6.id() }
+    pub fn connector_0_id(&self) -> ConnectorId { self.connectors.connector_0.id() }
```
Comment (franklynot) [inline src/bridge/graphs/peg_out.rs]: FYI, we do away with connector IDs in PR #78. They're no longer needed after the Winternitz secret rework.

File: src/bridge/graphs/peg_in.rs (hunk 2 above)
Comment (StarDumpling) [reply src/bridge/graphs/peg_in.rs]: sure, done

File: src/bridge/connectors/connector_0.rs (hunk 1 above)
Comment (StarDumpling) [reply src/bridge/connectors/connector_0.rs]: right, I removed `new_for_validation` fn from connectors I had modified earlier

**Review by franklynot — DISMISSED** [review]

**Review by franklynot — APPROVED** [review]

**Pull request merged**